pkg/serviceaccount/legacy.go

```diff
@@ -97,9 +104,9 @@ func (v *legacyValidator) Validate(tokenData string, public *jwt.Claims, private
 
 	if v.lookup {
 		// Make sure token hasn't been invalidated by deletion of the secret
-		secret, err := v.getter.GetSecret(namespace, secretName)
+		secret, err := v.getter.GetSecretWithMultiTenancy(tenant, namespace, secretName)
 		if err != nil {
-			klog.V(4).Infof("Could not retrieve token %s/%s for service account %s/%s: %v", namespace, secretName, namespace, serviceAccountName, err)
+			klog.V(4).Infof("Could not retrieve token %s/%s for service account %s/%s/%s: %v", namespace, secretName, tenant, namespace, serviceAccountName, err)
```

Comment:
Probably changing "token" to "secret" will make the message more accurate.

Reply:
fixed